### PR TITLE
Database column operations by fieldname

### PR DESCRIPTION
**Pull request by rustyrussell**
Our field number accesses leave a lot to be desired, particularly when changing queries.  After trying several approaches, this was the cleanest, and I think most readable.

(I was doing some db manipulations in another patch, and changing large SELECT statements is very painful to get right!).

Changelog-None: internal cleanup

### CODE REVIEW

File: wallet/db.c [hunk 1]
```diff
@@ -2165,6 +2165,323 @@ u8 *db_column_talarr(const tal_t *ctx, struct db_stmt *stmt, int col)
 			   db_column_bytes(stmt, col), 0);
 }
 
+/* Modern variants: by name */
+u64 db_col_u64(struct db_stmt *stmt, const char *colname)
+{
+	size_t col = db_query_colnum(stmt, colname);
+
+	if (db_column_is_null(stmt, col)) {
```
Comment (ZmnSCPxj) [inline wallet/db.c:2230]: Why is this more complex than `return db_column_u64(stmt, db_query_colnum(stmt, colname));`?  The existing `db_column_`* functions do the same checks.  Ditto for all the other `db_col_`* functions.

File: wallet/db.c [hunk 2]
```diff
@@ -934,14 +955,26 @@ struct db_stmt *db_prepare_v2_(const char *location, struct db *db,
...
+	}
+#endif
+	return ret;
 }
 
 u64 db_column_u64(struct db_stmt *stmt, int col)
 {
 	if (db_column_is_null(stmt, col)) {
 		log_broken(stmt->db->log, "Accessing a null column %d in query %s", col, stmt->query->query);
+		abort();
```
Comment (ZmnSCPxj) [inline wallet/db.c:987]: Why an abort here?

Comment (rustyrussell) [reply wallet/db.c:987]: Oops, left on from debugging!!!

Good catch

File: wallet/db.c (hunk 1 above)
Comment (rustyrussell) [reply wallet/db.c:2230]: I cut and paste: I'll be removing the old functions

Comment (rustyrussell) [reply wallet/db.c]: Done!

File: wallet/db.c (hunk 2 above)
Comment (rustyrussell) [reply wallet/db.c]: Fixed

**Review by vincenzopalazzo — APPROVED** [review]
ack https://github.com/ElementsProject/lightning/pull/4899/commits/f83cf8784888e0abf5d83a2b0e5ddb1d3d9fcf1d

failure GI is unrelated, and it looks like a `lnprotocoltest` failure, I will look into t[his failure](https://github.com/ElementsProject/lightning/runs/4160658040?check_suite_focus=true) and looks if we can avoid it

**Comment by rustyrussell** [comment]
Trivial rebase onto fresh master.

**Comment by ZmnSCPxj** [comment]
Kicked CI in case it is just test flakiness

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/5539d69e5baab4a191522e4cca66b2608cf7b138

after rebase

**Pull request merged**